spec/javascript/utils/form/validations.test.js

```diff
@@ -0,0 +1,56 @@
+import validate from 'utils/form/validations'
+
+// Validations will return this message if failed, null if passed
+const FAILED_VALIDATION = 'failed validation'
+
+describe('validate', () => {
+  describe('isOldEnough', () => {
+    test('passes validation if DOB is more than 18 years ago', () => {
+      expect(validate.isOldEnough(FAILED_VALIDATION)(['2000', '01', '12'])).toEqual(null)
+    })
+    test('fails validation if DOB is less than 18 years ago', () => {
+      expect(validate.isOldEnough(FAILED_VALIDATION)(['2010', '01', '12'])).toEqual(FAILED_VALIDATION)
+    })
+    test('fails validation if DOB is not a valid date', () => {
+      expect(validate.isOldEnough(FAILED_VALIDATION)(['2010', '13', '30'])).toEqual(FAILED_VALIDATION)
+    })
+  })
+  describe('isValidDate', () => {
+    test('passes validation if date is valid and 0-padded', () => {
+      expect(validate.isValidDate(FAILED_VALIDATION)(['2000', '02', '29'])).toEqual(null)
+    })
+    test('passes validation if date is valid and not 0-padded', () => {
+      expect(validate.isValidDate(FAILED_VALIDATION)(['2000', '2', '9'])).toEqual(null)
+    })
+    test('passes validation if date is valid and values are integers', () => {
+      expect(validate.isValidDate(FAILED_VALIDATION)([2000, 2, 9])).toEqual(null)
+    })
+    test('fails validation if date is a leap day in the wrong year', () => {
+      expect(validate.isValidDate(FAILED_VALIDATION)(['1999', '02', '29'])).toEqual(FAILED_VALIDATION)
+    })
+    test('fails validation if date is in the future', () => {
```

Comment:
same comment as above about being excessively proper and using dynamically generated dates

Reply:
fixed